src/Kmd.Logic.Gateway.Automation/PublishProducts.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Client;
+using Kmd.Logic.Gateway.Automation.PublishFile;
+using Kmd.Logic.Identity.Authorization;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    internal class PublishProducts
+    {
+        private readonly PublishPolicies _publishPolicies;
+        private readonly GatewayOptions options;
+        private readonly GatewayClientFactory gatewayClientFactory;
+        private List<GatewayAutomationResult> _publishResults;
+
+        public PublishProducts(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, GatewayOptions options, List<GatewayAutomationResult> publishResults)
```

Comment:
Why it has to be public

Reply:
Since it was a constructor kept it public, but now made internal, no harm.